sdk/python/feast/constants.py

```diff
@@ -21,135 +21,194 @@ class AuthProvider(Enum):
+    """
+
+    def __new__(cls, name, bases, attrs):
+        keys = [
+            k
+            for k, v in attrs.items()
+            if not k.startswith("_") and isinstance(v, (str, int, float))
+        ]
+        attrs["__config_keys__"] = keys
+        attrs.update({k: Option(k, attrs[k]) for k in keys})
+        return super().__new__(cls, name, bases, attrs)
+
+
+#: Default datetime column name for point-in-time join
+DATETIME_COLUMN: str = "datetime"
+
+#: Environmental variable to specify Feast configuration file location
+FEAST_CONFIG_FILE_ENV: str = "FEAST_CONFIG"
+
+#: Default prefix to Feast environmental variables
+CONFIG_FEAST_ENV_VAR_PREFIX: str = "FEAST_"
+
+#: Default directory to Feast configuration file
+CONFIG_FILE_DEFAULT_DIRECTORY: str = ".feast"
+
+#: Default Feast configuration file name
+CONFIG_FILE_NAME: str = "config"
+
+#: Default section in Feast configuration file to specify options
+CONFIG_FILE_SECTION: str = "general"
+
+
+class ConfigOptions(metaclass=ConfigMeta):
+    """ Feast Configuration Options """
+
+    #: Feast project namespace to use
+    PROJECT: str = "default"
+
+    #: Default Feast Core URL
+    CORE_URL: str = "localhost:6565"
+
+    #: Enable or disable TLS/SSL to Feast Core
+    CORE_ENABLE_SSL: str = "False"
+
+    #: Enable user authentication to Feast Core
+    ENABLE_AUTH: str = "False"
+
+    #: Auth token for user authentication to Feast
+    AUTH_TOKEN: str = ""
+
+    #: Path to certificate(s) to secure connection to Feast Core
+    CORE_SERVER_SSL_CERT: str = ""
+
+    #: Default Feast Serving URL
+    SERVING_URL: str = "localhost:6566"
+
+    #: Enable or disable TLS/SSL to Feast Serving
+    SERVING_ENABLE_SSL: str = "False"
+
+    #: Path to certificate(s) to secure connection to Feast Serving
+    SERVING_SERVER_SSL_CERT: str = ""
+
+    #: Default Feast Job Service URL
+    JOB_SERVICE_URL: str = ""
+
+    #: Enable or disable TLS/SSL to Feast Job Service
+    JOB_SERVICE_ENABLE_SSL: str = "False"
+
+    #: Path to certificate(s) to secure connection to Feast Job Service
+    JOB_SERVICE_SERVER_SSL_CERT: str = ""
+
+    #: Enable or disable control loop for Feast Job Service
+    JOB_SERVICE_ENABLE_CONTROL_LOOP: str = "False"
+
+    #: Default connection timeout to Feast Serving, Feast Core, and Feast Job Service (in seconds)
+    GRPC_CONNECTION_TIMEOUT: str = "10"
+
+    #: Default gRPC connection timeout when sending an ApplyFeatureSet command to Feast Core (in seconds)
+    GRPC_CONNECTION_TIMEOUT_APPLY: str = "600"
+
+    #: Default timeout when running batch ingestion
+    BATCH_INGESTION_PRODUCTION_TIMEOUT: str = "120"
+
+    #: Time to wait for historical feature requests before timing out.
+    BATCH_FEATURE_REQUEST_WAIT_TIME_SECONDS: str = "600"
+
+    #: Authentication Provider - Google OpenID/OAuth
+    AUTH_PROVIDER: str = "google"
+
+    #: Spark Job launcher
+    SPARK_LAUNCHER: str = "dataproc"  # standalone, dataproc, emr
```

Comment:
Should we change this to `standalone` so that users arent forced to use dataproc by default?

Reply:
Defaults to None. Users should set this on their own.